examples/audio/uk_ireland_accent_recognition.py

```diff
@@ -0,0 +1,664 @@
+
+"""
+## Callbacks
+
+A couple of Keras callbacks in order to:
+*   Stop whenever the validation AUC stops improving
+*   Save the best model
+*   Call TensorBoard in order to later view the training and validation logs
+"""
+
+early_stopping_cb = keras.callbacks.EarlyStopping(
+    monitor="val_auc", patience=10, restore_best_weights=True
+)
+
+model_checkpoint_cb = keras.callbacks.ModelCheckpoint(
+    MODEL_NAME + ".h5", monitor="val_auc", save_best_only=True
+)
+
+tensorboard_cb = keras.callbacks.TensorBoard(
+    os.path.join(os.curdir, "logs", model.name)
+)
+
+callbacks = [early_stopping_cb, model_checkpoint_cb, tensorboard_cb]
+
+"""
+## Training
+"""
+
+history = model.fit(
+    train_ds,
+    epochs=EPOCHS,
+    validation_data=valid_ds,
+    class_weight=class_weight,
+    callbacks=callbacks,
+    verbose=2,
+)
+
+"""
+## Results
+"""
+
+"""
+### TensorBoard
+"""
+
+"""
+```
+%reload_ext tensorboard
+
+%tensorboard --logdir=./logs/ --port=6066
+```
+"""
+
+"""
+### Evaluation
+"""
+
+train_loss, train_acc, train_auc = model.evaluate(train_ds)
+valid_loss, valid_acc, valid_auc = model.evaluate(valid_ds)
+
+print(
+    "train d-prime: {0:.3f}, validation d-prime: {1:.3f}".format(
+        d_prime(train_auc), d_prime(valid_auc)
+    )
+)
+
+"""
+We can see that the model achieves the following results:
+
+Results    | Training  | Validation
+-----------|-----------|------------
+Accuracy   | 54%       | 51%
+AUC        | 0.9091    | 0.8910
+d-prime    | 1.888     | 1.742
+
+"""
+
+"""
+### Confusion Matrix
+
+Let's now plot the confusion matrix for the validation dataset.
+"""
+
+# Create X and y tensors
+X_valid = None
+y_valid = None
+
+for X, y in iter(valid_ds):
+    if X_valid is None:
+        X_valid = X.numpy()
+        y_valid = y.numpy()
+    else:
+        X_valid = np.concatenate((X_valid, X.numpy()), axis=0)
+        y_valid = np.concatenate((y_valid, y.numpy()), axis=0)
+
+# Generate predictions
+y_pred = model.predict(X_valid)
+
+# Calculate confusion matrix
+confusion_mtx = tf.math.confusion_matrix(
+    np.argmax(y_valid, axis=1), np.argmax(y_pred, axis=1)
+)
+
+# Plot the confusion matrix
+plt.figure(figsize=(10, 8))
+sns.heatmap(
+    confusion_mtx, xticklabels=class_names, yticklabels=class_names, annot=True, fmt="g"
+)
+plt.xlabel("Prediction")
+plt.ylabel("Label")
+plt.title("Validation Confusion Matrix")
+plt.show()
+
+"""
+### Precision & Recall
+"""
+
+for i, label in enumerate(class_names):
+    precision = confusion_mtx[i, i] / np.sum(confusion_mtx[:, i])
+    recall = confusion_mtx[i, i] / np.sum(confusion_mtx[i, :])
+    print(
+        "{0:15} Precision:{1:.2f}%; Recall:{2:.2f}%".format(
+            label, precision * 100, recall * 100
+        )
+    )
+
+"""
+# Inference
+"""
+
+"""
+The below function `yamnet_class_names_from_csv` was copied and very slightly changed
+from [Yamnet
+Notebook](https://colab.research.google.com/github/tensorflow/hub/blob/master/examples/colab/yamnet.ipynb)
```

Comment:
Make sure to format all markdown links on a single line (see the guide: https://github.com/keras-team/keras-io/blob/master/contributor_guide.md)